lib/core/src/miscUtil.cpp

```diff
@@ -248,6 +249,7 @@ getRodsObjType( rcComm_t *conn, rodsPath_t *rodsPath ) {
             rstrcpy( rodsPath->chksum, rodsObjStatOut->chksum, NAME_LEN );
         }
     }
+    std::free(rodsPath->rodsObjStat);
```

Comment:
I take it `rodsPath->rodsObjStat` is always valid at this point?

Should it be free'd on lines 221 and 226 too?

Reply:
Since the next line replaces `rodsPath->rodsObjStat`, we need to free the value held there. The other returns do not override `rodsPath->rodsObjStat`.

However, `rodsObjStatOut` _might_ leak in those cases. If `rcObjStat(...)` still creates a `rodsObjStat_t` when it returns a `status < 0`, then it may leak.

Reply:
> Should we check the pointer before freeing it?

The only check I can think of is to check if `rodsPath->rodsObjStat == rodsObjStatOut`? I have no idea if `rcObjStat(...)` would return it's input like that...

Otherwise, if it's a `nullptr`, there is no need to check as free is safe with `nullptr`. If there's a value that needs to be freed, it will be freed. If it's a pointer to an already freed structure, there's no check for that, and it's UB (catchable by ASAN).

Reply:
Got it.